Index: dbt7/DSGen-software-code-4.0.0_final/tools/config.h
===================================================================
--- dbt7.orig/DSGen-software-code-4.0.0_final/tools/config.h
+++ dbt7/DSGen-software-code-4.0.0_final/tools/config.h
@@ -99,9 +99,10 @@
 #endif /* INTERIX */
 
 #ifdef LINUX
+#include <inttypes.h>
 #define SUPPORT_64BITS
 #define HUGE_TYPE	int64_t
-#define HUGE_FORMAT	"%lld"
+#define HUGE_FORMAT	"%" PRId64
 #define HUGE_COUNT	1
 #define USE_STRING_H
 #define USE_VALUES_H
Index: dbt7/DSGen-software-code-4.0.0_final/tools/s_zip_to_gmt.c
===================================================================
--- dbt7.orig/DSGen-software-code-4.0.0_final/tools/s_zip_to_gmt.c
+++ dbt7/DSGen-software-code-4.0.0_final/tools/s_zip_to_gmt.c
@@ -149,7 +149,7 @@ mk_s_zip(void *pDest, ds_key_t kIndex)
 	pMap = &arGMTOffset[0];
 	while (kIndex > pMap->nMax)
 		pMap += 1;
-	sprintf(r->szZip, "%05lld", kIndex);
+	snprintf(r->szZip, sizeof(r->szZip), HUGE_FORMAT, kIndex);
 	r->nGMTOffset = pMap->nOffset;
 	
 	return(r->nGMTOffset == -99?1:0);
Index: dbt7/DSGen-software-code-4.0.0_final/tools/date.c
===================================================================
--- dbt7.orig/DSGen-software-code-4.0.0_final/tools/date.c
+++ dbt7/DSGen-software-code-4.0.0_final/tools/date.c
@@ -563,7 +563,7 @@ day_number(date_t *d)
  * TODO: None
  */
 int
-getDateWeightFromJulian(jDay, nDistribution)
+getDateWeightFromJulian(int jDay, int nDistribution)
 {
 	date_t dTemp;
 	int nDay;
Index: dbt7/DSGen-software-code-4.0.0_final/tools/tdef_functions.c
===================================================================
--- dbt7.orig/DSGen-software-code-4.0.0_final/tools/tdef_functions.c
+++ dbt7/DSGen-software-code-4.0.0_final/tools/tdef_functions.c
@@ -171,7 +171,7 @@ table_func_t s_tdef_funcs[] = {
 };
 
 table_func_t *
-getTdefFunctionsByNumber(nTable)
+getTdefFunctionsByNumber(int nTable)
 {
    if (nTable >= S_BRAND)
       return(&s_tdef_funcs[nTable - S_BRAND]);
Index: dbt7/DSGen-software-code-4.0.0_final/tools/tdefs.c
===================================================================
--- dbt7.orig/DSGen-software-code-4.0.0_final/tools/tdefs.c
+++ dbt7/DSGen-software-code-4.0.0_final/tools/tdefs.c
@@ -147,7 +147,7 @@ getTdefsByNumber(int nTable)
 }
 */
 tdef *
-getSimpleTdefsByNumber(nTable)
+getSimpleTdefsByNumber(int nTable)
 {
    if (nTable >= S_BRAND)
       return(&s_tdefs[nTable - S_BRAND]);
