Index: dbt7/DSGen-software-code-3.2.0rc1/tools/tpcds.dst
===================================================================
--- dbt7.orig/DSGen-software-code-3.2.0rc1/tools/tpcds.dst
+++ dbt7/DSGen-software-code-3.2.0rc1/tools/tpcds.dst
@@ -646,7 +646,7 @@ add ("COMOROS":1);
 --add ("CONGO, THE DEMOCRATIC REPUBLIC OF THE":1);
 add ("add ("COOK ISLANDS":1);
 add ("COSTA RICA":1);
-add ("CÔTE D'IVOIRE":1);
+add ("CÃ”TE D'IVOIRE":1);
 add ("CROATIA":1);
 add ("CUBA":1);
 add ("CYPRUS":1);
@@ -771,7 +771,7 @@ add ("POLAND":1);
 add ("PORTUGAL":1);
 add ("PUERTO RICO":1);
 add ("QATAR":1);
-add ("RÉUNION":1);
+add ("RÃ‰UNION":1);
 add ("ROMANIA":1);
 add ("RUSSIAN FEDERATION":1);
 add ("RWANDA":1);
Index: dbt7/DSGen-software-code-3.2.0rc1/tools/tpcds.wam
===================================================================
--- dbt7.orig/DSGen-software-code-3.2.0rc1/tools/tpcds.wam
+++ dbt7/DSGen-software-code-3.2.0rc1/tools/tpcds.wam
@@ -5616,7 +5616,7 @@ D_DAY_NAME
 d_quarter_name
 D_QUARTER_NAME
 
-Name of the quarter (unique) Q199, Q299, …Q405\
+Name of the quarter (unique) Q199, Q299, Q405\
 dependent on quarter
 '#Qyy' where # is in (1,2,3,4), and yy is derived from D_YEAR.\
 This is a calendar quarter. This is NOT a FY quarter.
@@ -31362,4 +31362,3 @@ OLE     BIN
 
 /END_TABLE AMCSUBS
 
-
\ No newline at end of file
