Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query1.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query1.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query10.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query10.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query11.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query11.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query13.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query13.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query14.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query14.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query15.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query15.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query16.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query16.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query17.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query17.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query18.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query18.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query19.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query19.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query2.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query2.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query20.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query20.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query21.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query21.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query22.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query22.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query23.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query23.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query24.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query24.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query25.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query25.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query26.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query26.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query27.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query27.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query28.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query28.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query29.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query29.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query3.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query3.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query30.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query30.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query31.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query31.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query32.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query32.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query33.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query33.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query34.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query34.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query35.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query35.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query36.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query36.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query37.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query37.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query38.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query38.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query39.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query39.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query4.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query4.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query40.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query40.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query41.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query41.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query42.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query42.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query43.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query43.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query44.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query44.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query45.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query45.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query46.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query46.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query47.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query47.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query48.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query48.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query49.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query49.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query5.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query5.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query50.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query50.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query51.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query51.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query52.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query52.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query53.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query53.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query54.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query54.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query55.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query55.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query56.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query56.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query57.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query57.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query58.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query58.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query59.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query59.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query6.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query6.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query60.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query60.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query61.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query61.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query62.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query62.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query63.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query63.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query64.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query64.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query65.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query65.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query66.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query66.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query67.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query67.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query68.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query68.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query69.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query69.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query7.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query7.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query70.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query70.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query71.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query71.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query72.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query72.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query73.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query73.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query74.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query74.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query75.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query75.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query76.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query76.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query77.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query77.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query78.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query78.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query79.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query79.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query8.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query8.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query80.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query80.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query81.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query81.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query82.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query82.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query83.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query83.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query84.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query84.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query85.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query85.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query86.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query86.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query87.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query87.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query88.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query88.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query89.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query89.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query9.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query9.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query90.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query90.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query91.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query91.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query92.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query92.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query93.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query93.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query94.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query94.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query95.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query95.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query96.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query96.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query97.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query97.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query98.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query98.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query99.tpl
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_postgresql_test/query99.tpl
@@ -0,0 +1 @@
+SELECT pg_sleep(1);
Index: dbt7/DSGen-software-code-4.0.0_final/query_templates/templates-pgsql-test.lst
===================================================================
--- /dev/null
+++ dbt7/DSGen-software-code-4.0.0_final/query_templates/templates-pgsql-test.lst
@@ -0,0 +1,99 @@
+../query_postgresql_test/query1.tpl
+../query_postgresql_test/query2.tpl
+../query_postgresql_test/query3.tpl
+../query_postgresql_test/query4.tpl
+../query_postgresql_test/query5.tpl
+../query_postgresql_test/query6.tpl
+../query_postgresql_test/query7.tpl
+../query_postgresql_test/query8.tpl
+../query_postgresql_test/query9.tpl
+../query_postgresql_test/query10.tpl
+../query_postgresql_test/query11.tpl
+../query_postgresql_test/query12.tpl
+../query_postgresql_test/query13.tpl
+../query_postgresql_test/query14.tpl
+../query_postgresql_test/query15.tpl
+../query_postgresql_test/query16.tpl
+../query_postgresql_test/query17.tpl
+../query_postgresql_test/query18.tpl
+../query_postgresql_test/query19.tpl
+../query_postgresql_test/query20.tpl
+../query_postgresql_test/query21.tpl
+../query_postgresql_test/query22.tpl
+../query_postgresql_test/query23.tpl
+../query_postgresql_test/query24.tpl
+../query_postgresql_test/query25.tpl
+../query_postgresql_test/query26.tpl
+../query_postgresql_test/query27.tpl
+../query_postgresql_test/query28.tpl
+../query_postgresql_test/query29.tpl
+../query_postgresql_test/query30.tpl
+../query_postgresql_test/query31.tpl
+../query_postgresql_test/query32.tpl
+../query_postgresql_test/query33.tpl
+../query_postgresql_test/query34.tpl
+../query_postgresql_test/query35.tpl
+../query_postgresql_test/query36.tpl
+../query_postgresql_test/query37.tpl
+../query_postgresql_test/query38.tpl
+../query_postgresql_test/query39.tpl
+../query_postgresql_test/query40.tpl
+../query_postgresql_test/query41.tpl
+../query_postgresql_test/query42.tpl
+../query_postgresql_test/query43.tpl
+../query_postgresql_test/query44.tpl
+../query_postgresql_test/query45.tpl
+../query_postgresql_test/query46.tpl
+../query_postgresql_test/query47.tpl
+../query_postgresql_test/query48.tpl
+../query_postgresql_test/query49.tpl
+../query_postgresql_test/query50.tpl
+../query_postgresql_test/query51.tpl
+../query_postgresql_test/query52.tpl
+../query_postgresql_test/query53.tpl
+../query_postgresql_test/query54.tpl
+../query_postgresql_test/query55.tpl
+../query_postgresql_test/query56.tpl
+../query_postgresql_test/query57.tpl
+../query_postgresql_test/query58.tpl
+../query_postgresql_test/query59.tpl
+../query_postgresql_test/query60.tpl
+../query_postgresql_test/query61.tpl
+../query_postgresql_test/query62.tpl
+../query_postgresql_test/query63.tpl
+../query_postgresql_test/query64.tpl
+../query_postgresql_test/query65.tpl
+../query_postgresql_test/query66.tpl
+../query_postgresql_test/query67.tpl
+../query_postgresql_test/query68.tpl
+../query_postgresql_test/query69.tpl
+../query_postgresql_test/query70.tpl
+../query_postgresql_test/query71.tpl
+../query_postgresql_test/query72.tpl
+../query_postgresql_test/query73.tpl
+../query_postgresql_test/query74.tpl
+../query_postgresql_test/query75.tpl
+../query_postgresql_test/query76.tpl
+../query_postgresql_test/query77.tpl
+../query_postgresql_test/query78.tpl
+../query_postgresql_test/query79.tpl
+../query_postgresql_test/query80.tpl
+../query_postgresql_test/query81.tpl
+../query_postgresql_test/query82.tpl
+../query_postgresql_test/query83.tpl
+../query_postgresql_test/query84.tpl
+../query_postgresql_test/query85.tpl
+../query_postgresql_test/query86.tpl
+../query_postgresql_test/query87.tpl
+../query_postgresql_test/query88.tpl
+../query_postgresql_test/query89.tpl
+../query_postgresql_test/query90.tpl
+../query_postgresql_test/query91.tpl
+../query_postgresql_test/query92.tpl
+../query_postgresql_test/query93.tpl
+../query_postgresql_test/query94.tpl
+../query_postgresql_test/query95.tpl
+../query_postgresql_test/query96.tpl
+../query_postgresql_test/query97.tpl
+../query_postgresql_test/query98.tpl
+../query_postgresql_test/query99.tpl
