Index: dbt7/DSGen-software-code-3.2.0rc1/tools/driver.c
===================================================================
--- dbt7.orig/DSGen-software-code-3.2.0rc1/tools/driver.c
+++ dbt7/DSGen-software-code-3.2.0rc1/tools/driver.c
@@ -398,6 +398,7 @@ main (int ac, char **av)
 	struct timeb t;
    tdef *pT;
    table_func_t *pF;
+	extern int this_is_the_table_to_stream;
 	
 	process_options (ac, av);
 	validate_options();
@@ -560,6 +561,7 @@ main (int ac, char **av)
          /*
          * now build the actual rows
          */
+         this_is_the_table_to_stream = i;
          gen_tbl(i, kFirstRow, kRowCount);
 		}
 	}
Index: dbt7/DSGen-software-code-3.2.0rc1/tools/params.h
===================================================================
--- dbt7.orig/DSGen-software-code-3.2.0rc1/tools/params.h
+++ dbt7/DSGen-software-code-3.2.0rc1/tools/params.h
@@ -61,7 +61,7 @@ option_t options[] =
 {"CHILD",		OPT_INT,			17,"generate <n>th chunk of the parallelized data", NULL, "1"}, 
 {"CHKSEEDS",		OPT_FLG|OPT_HIDE,	18, "validate RNG usage for parallelism", NULL, "N"}, 
 {"RELEASE",		OPT_FLG,			19, "display the release information", printReleaseInfo, "N"}, 
-{"_FILTER",		OPT_FLG,			20, "output data to stdout", NULL, "N"}, 
+{"FILTER",		OPT_FLG,			20, "output data to stdout", NULL, "N"}, 
 {"VALIDATE",	OPT_FLG,			21, "produce rows for data validation", NULL, "N"}, 
 {"VCOUNT",		OPT_INT|OPT_ADV,	22, "set number of validation rows to be produced", NULL, "50"}, 
 {"VSUFFIX",		OPT_STR|OPT_ADV,	23, "set file suffix for data validation", NULL, ".vld"}, 
Index: dbt7/DSGen-software-code-3.2.0rc1/tools/print.c
===================================================================
--- dbt7.orig/DSGen-software-code-3.2.0rc1/tools/print.c
+++ dbt7/DSGen-software-code-3.2.0rc1/tools/print.c
@@ -57,6 +57,7 @@ static FILE *fpDeleteFile;
 static char *arDeleteFiles[3] = {"", "delete_", "inventory_delete_"};
 
 static int current_table = -1;
+int this_is_the_table_to_stream = -1;
 
 int print_jdate (FILE *pFile, ds_key_t kValue);
 
@@ -65,6 +66,9 @@ print_close(int tbl)
 {
    tdef *pTdef = getSimpleTdefsByNumber(tbl);
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	fpOutfile = NULL;
 	if (pTdef->outfile)
 	{
@@ -85,6 +89,9 @@ print_separator (int sep)
 	static char *pDelimiter;
 	static int init = 0;
 	
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return (0);
+
 	if (!init)
 	{
         pDelimiter = get_str ("DELIMITER");
@@ -141,6 +148,9 @@ print_prep (int table, int update)
 void
 print_integer (int nColumn, int val, int sep)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn))
 	{
 		if (fprintf (fpOutfile, "%d", val) < 0)
@@ -162,6 +172,9 @@ print_varchar (int nColumn, char *val, i
 {
 	size_t nLength;
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn) && (val != NULL))
 	{
       nLength = strlen(val);
@@ -189,6 +202,9 @@ print_varchar (int nColumn, char *val, i
 void
 print_delete (int * val)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
    if (print_jdate(fpDeleteFile, *val))
 	{
       fprintf(stderr, "ERROR: Failed to write delete key\n");
@@ -232,6 +248,9 @@ print_cp_delete (int nCatalog, int nPage
 void
 print_char (int nColumn, char val, int sep)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn))
 	{
 		if (fwrite (&val, 1, 1, fpOutfile) != 1)
@@ -251,6 +270,9 @@ print_char (int nColumn, char val, int s
 void
 print_date (int nColumn, ds_key_t val, int sep)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn))
 	{
 		if (val > 0)
@@ -280,6 +302,9 @@ print_time (int nColumn, ds_key_t val, i
 {
 	int nHours, nMinutes, nSeconds;
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	nHours = (int)(val / 3600);
 	val -= 3600 * nHours;
 	nMinutes = (int)(val / 60);
@@ -311,6 +336,9 @@ print_decimal (int nColumn, decimal_t *
 	int i;
 	double dTemp;
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn))
 	{
 #ifdef WIN32
@@ -340,6 +368,9 @@ print_decimal (int nColumn, decimal_t *
 void
 print_key (int nColumn, ds_key_t val, int sep)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn))
 	{
 		if (val != (ds_key_t) -1) /* -1 is a special value, indicating NULL */
@@ -399,6 +430,9 @@ print_id (int nColumn, ds_key_t val, int
 void
 print_boolean (int nColumn, int val, int sep)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!nullCheck(nColumn))
 	{
 
@@ -479,10 +513,10 @@ print_start (int tbl)
 #else
 		   pTdef->outfile = fopen (path, "w");
 #endif
+		   fpOutfile = pTdef->outfile;
 	   }
    }
    
-   fpOutfile = pTdef->outfile;
    res = (fpOutfile != NULL);
 
    if (!res)                    /* open failed! */
@@ -524,6 +558,9 @@ print_end (int tbl)
    static int add_term = 0;
    static char term[10];
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return (0);
+
    if (!init)
      {
         if (is_set ("TERMINATE"))
@@ -562,6 +599,9 @@ openDeleteFile(int bOpen)
    int res = 0;
    char path[256];
    
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return (0);
+
    if (!bOpen)
       fclose(fpDeleteFile);
    else
@@ -615,6 +655,9 @@ openDeleteFile(int bOpen)
 void
 print_string (char *szMessage, ds_key_t val)
 {
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (fprintf (fpOutfile, szMessage, val) < 0)
 	{
 		fprintf(stderr, "ERROR: Failed to write string\n");
@@ -643,6 +686,9 @@ print_jdate (FILE *pFile, ds_key_t kValu
 
 	date_t dTemp;
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return (0);
+
 	jtodt (&dTemp, (int) kValue);
 #if (defined(STR_QUOTES) && !defined(_MYSQL))
 	if ((fwrite ("\"", 1, 1, pFile) != 1) ||
@@ -675,6 +721,9 @@ print_validation(ds_key_t kRowNumber)
 	static int bInit = 0;
 	static char szValidateFormat[20];
 
+	if (is_set("FILTER") && current_table != this_is_the_table_to_stream)
+		return;
+
 	if (!bInit)
 	{
 		sprintf(szValidateFormat, "Row #%s: ", HUGE_FORMAT);
