Index: dbt7/dsgen/tools/build_support.c
===================================================================
--- dbt7.orig/dsgen/tools/build_support.c
+++ dbt7/dsgen/tools/build_support.c
@@ -369,7 +369,7 @@ embed_string(char *szDest, char *szDist,
 
 	pick_distribution(&szWord, szDist, nValue, nWeight, nStream);
 	nPosition = genrand_integer(NULL, DIST_UNIFORM, 0, strlen(szDest) - strlen(szWord) - 1, 0, nStream);
-	strncpy(&szDest[nPosition], szWord, nStream);
+	strncpy(&szDest[nPosition], szWord, strlen(&szDest[nPosition]));
 
 	return(0);
 }
